File: frontend/src/types/interviewInvitation.ts
import type { InterviewAnswerSubmitRequest, InterviewRuntime } from './interview'

export interface InterviewInvitationCreateRequest {
  email: string
  expires_in_days: number
  max_access_count: number
}

export interface InterviewInvitation {
  id: string
  interview_session_id: string
  email: string
  status: string
  max_access_count: number
  access_count: number
  expires_at: string
  opened_at: string | null
  verified_at: string | null
  consented_at: string | null
  started_at: string | null
  completed_at: string | null
  revoked_at: string | null
  created_at: string
  invitation_token: string | null
  access_code: string | null
}

export interface InterviewInvitationCreated extends InterviewInvitation {
  invitation_token: string
  access_code: string
}

export interface PublicInterviewInvitation {
  invitation_id: string
  workspace_name: string
  job_title: string
  candidate_name: string
  masked_email: string
  scheduled_at: string | null
  expires_at: string
  status: string
  evaluation_status: 'PENDING' | 'GENERATING' | 'COMPLETED' | 'FAILED' | null
  decision: 'HIRED' | 'REJECTED' | null
  decided_at: string | null
}

export interface InterviewInvitationVerifyRequest {
  email: string
  access_code: string
  consent: boolean
}

export interface CandidateInterviewAccess {
  invitation_id: string
  interview_session_id: string
  access_token: string
  token_type: 'candidate_interview'
  expires_at: string
}

export type { InterviewAnswerSubmitRequest, InterviewRuntime }

File: backend/app/schemas/interview_invitation.py
import uuid
from datetime import datetime

from pydantic import BaseModel, EmailStr, Field, field_validator


class InterviewInvitationCreateRequest(BaseModel):
    email: EmailStr
    expires_in_days: int = Field(default=7, ge=1, le=30)
    max_access_count: int = Field(default=5, ge=1, le=20)

    @field_validator("email", mode="before")
    @classmethod
    def normalize_email(cls, value: str) -> str:
        return value.strip().lower()


class InterviewInvitationResponse(BaseModel):
    id: uuid.UUID
    interview_session_id: uuid.UUID
    email: EmailStr
    status: str
    max_access_count: int
    access_count: int
    expires_at: datetime
    opened_at: datetime | None
    verified_at: datetime | None
    consented_at: datetime | None
    started_at: datetime | None
    completed_at: datetime | None
    revoked_at: datetime | None
    created_at: datetime
    invitation_token: str | None = None
    access_code: str | None = None


class InterviewInvitationCreateResponse(InterviewInvitationResponse):
    invitation_token: str
    access_code: str


class PublicInterviewInvitationResponse(BaseModel):
    invitation_id: uuid.UUID
    workspace_name: str
    job_title: str
    candidate_name: str
    masked_email: str
    scheduled_at: datetime | None
    expires_at: datetime
    status: str
    evaluation_status: str | None = None
    decision: str | None = None
    decided_at: datetime | None = None


class InterviewInvitationVerifyRequest(BaseModel):
    email: EmailStr
    access_code: str = Field(min_length=6, max_length=12)
    consent: bool

    @field_validator("email", mode="before")
    @classmethod
    def normalize_email(cls, value: str) -> str:
        return value.strip().lower()

    @field_validator("access_code")
    @classmethod
    def normalize_access_code(cls, value: str) -> str:
        return value.strip()


class InterviewCandidateAccessResponse(BaseModel):
    invitation_id: uuid.UUID
    interview_session_id: uuid.UUID
    access_token: str
    token_type: str = "candidate_interview"
    expires_at: datetime
